test/lib/validation/form/deprecated-appearance.spec.js

```diff
@@ -0,0 +1,169 @@
+const { expect } = require('chai');
+const { DOMParser } = require('@xmldom/xmldom');
+const deprecatedAppearance = require('../../../../src/lib/validation/form/deprecated-appearance');
+
+const domParser = new DOMParser();
+
+const getXml = ({ name = '', age = '', address = '', streetNbr = '', city = '' }) => `
+<?xml version="1.0"?>
+<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms/" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <h:head>
+    <h:title>Test</h:title>
+    <model>
+      <instance>
+        <data id="ABC" version="2015-06-05">
+          <name>Harambe</name>
+          <age/>
+          <address>
+            <street-nbr/>
+            <city/>
+            <appearance/>
+          </address>
+        </data>
+      </instance>
+      <bind nodeset="/data/name" type="string" />   
+      <meta>
+        <instanceID/>
+      </meta>
+    </model>
+  </h:head>
+  <h:body>
+    <input ref="/data/name" appearance="${name}">
+      <label>What is the name?</label>
+    </input>
+    <input ref="/data/age" appearance="${age}">
+      <label>What is the age?</label>
+    </input>
+    <group ref="/data/address" appearance="${address}">
+        <input appearance="${streetNbr}" ref="/data/address/street-nbr">
+          <label>What is the address?</label>
+        </input>
+        <input appearance="${city}" ref="/data/address/city">
+          <label>What is the city?</label>
+        </input>
+        <input ref="/data/address/appearance">
+          <label>What is the appearance?</label>
+        </input>
+      </group>
+  </h:body>
+</h:html>`;
+
+const getXmlDoc = (appearances = {}) => domParser.parseFromString(getXml(appearances));
+const xformPath = '/my/form/path/form.xml';
+
+const assertEmpty = (output) => {
+  expect(output.warnings).is.empty;
+  expect(output.errors, output.errors).is.empty;
+};
+
+const LATEST_VERSION = '999.99.99';
+const ERROR_HEADER = `Form at ${xformPath} contains fields with the deprecated \`horizontal\`/\`compact\` appearance. ` +
+  'These have been deprecated in favor of the `columns` appearance. Please update the following fields:';
+
+describe('deprecated-appearance', () => {
+  it(`resolves OK for form with no appearances`, () => {
+    return deprecatedAppearance.execute({ xformPath, xmlDoc: getXmlDoc(), apiVersion: LATEST_VERSION })
```

Comment:
```suggestion
    return deprecatedAppearance
      .execute({ xformPath, xmlDoc: getXmlDoc(), apiVersion: LATEST_VERSION })
```
We usually put the chain functions in separate lines. However we don't always enforce it for unit tests. Leaving it here as op

Reply:
actually you can also await, right? 
```
const output = await deprecatedAppearance.execute(....);
assertEmpty(output);
```

Reply:
+1 move putting the chain functions on separate lines.  

Honestly I am torn on the `await` vs promise chaining.  Using `await` seems more straight forward, but I have grown really attached to the clean flow of promise chaining!  :heart: 

I see both styles throughout the code base (well less `await` probably since it is a newer feature), but if there is a reason to prefer one over the other in newly added code, I am happy to use whichever...

Reply:
Both are okay, I prefer `await` when `then()` content has too many levels inside (likely a code smell) and helps flatting code for readability. 
These `then()` are easy and clear, it's okay